### ISSUE TITLE
Validation loss/accuracy inconsistency problem

### ISSUE THREAD

**Issue by bearpaw** [anewell/pose-hg-train#19]
Hi @anewell ,

I found that during training, I can get the following log
```
==> Starting epoch: 40/100
 [======================================== 8000/8000 ==================================>]  Tot: 1h10m | Step: 526ms     
      train : Loss: 0.0056964 Acc: 0.8067
 [======================================== 1000/1000 ==================================>]  Tot: 1m54s | Step: 116ms     
      valid : Loss: 0.0060296 Acc: 0.8033
```

Then I use the following cmd to do testing on validation set
```
th main.lua -dataDir ../data/ -expDir ../exp/ -trainIters 0 -loadModel ../exp/mpii/default/model_40.t7
```
and get the following result! The validation accuracy and loss are different with those during training.
```
Saving everything to: /home/wyang/code/pose/iccv17/pose-hg-train/exp/mpii/default	
Input is a tensor with dimensions: 3 x 256 x 256	
Output is a table	
	 Entry 1 is a tensor with dimensions: 16 x 64 x 64	
	 Entry 2 is a tensor with dimensions: 16 x 64 x 64	
	 Entry 3 is a tensor with dimensions: 16 x 64 x 64	
	 Entry 4 is a tensor with dimensions: 16 x 64 x 64	
	 Entry 5 is a tensor with dimensions: 16 x 64 x 64	
	 Entry 6 is a tensor with dimensions: 16 x 64 x 64	
	 Entry 7 is a tensor with dimensions: 16 x 64 x 64	
	 Entry 8 is a tensor with dimensions: 16 x 64 x 64	
==> Loading model from: ../exp/mpii/default/model_40.t7	
==> Converting model to CUDA	
==> Starting epoch: 1/100	
 [======================================== 1000/1000 ==================================>]  Tot: 2m42s | Step: 115ms     
      valid : Loss: 0.0050312 Acc: 0.8510	
```

I check your code many times but still have no idea. Can you repeat my problem?  How to solve this? Thank you.

**Comment by anewell** [comment]
It is because the code generates a random training/validation split if you start it from scratch. If you use the -branch option to continue an experiment then it will keep the original split otherwise the random split will put some images that it was trained on into the mix.

**Comment by bearpaw** [comment]
@anewell Thanks! I checked the code and found the train/val split is computed in https://github.com/anewell/pose-hg-train/blob/master/src/util/dataset/mpii.lua#L35-L36
